crates/router/src/connector/cashtocode/transformers.rs

```diff
@@ -0,0 +1,228 @@
+                api_key: api_key.to_string(),
+            }),
+            _ => Err(errors::ConnectorError::FailedToObtainAuthType.into()),
+        }
+    }
+}
+
+#[derive(Debug, Clone, Default, Serialize, Deserialize)]
+#[serde(rename_all = "lowercase")]
+pub enum CashtocodePaymentStatus {
+    Succeeded,
+    Failed,
+    #[default]
+    Processing,
+}
+
+impl From<CashtocodePaymentStatus> for enums::AttemptStatus {
+    fn from(item: CashtocodePaymentStatus) -> Self {
+        match item {
+            CashtocodePaymentStatus::Succeeded => Self::Charged,
+            CashtocodePaymentStatus::Failed => Self::Failure,
+            CashtocodePaymentStatus::Processing => Self::Authorizing,
+        }
+    }
+}
+
+#[derive(Debug, Deserialize, Clone)]
+pub struct CashtocodeErrors {
+    pub message: String,
+    pub path: String,
+    #[serde(rename = "type")]
+    pub event_type: String,
+}
+
+#[derive(Debug, Clone, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct CashtocodePaymentsResponse {
+    pub pay_url: String,
+}
+
+pub struct CashtocodePaymentsSyncResponse {}
+
+impl<F, T>
+    TryFrom<
+        types::ResponseRouterData<F, CashtocodePaymentsResponse, T, types::PaymentsResponseData>,
+    > for types::RouterData<F, T, types::PaymentsResponseData>
+{
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(
+        item: types::ResponseRouterData<
+            F,
+            CashtocodePaymentsResponse,
+            T,
+            types::PaymentsResponseData,
+        >,
+    ) -> Result<Self, Self::Error> {
+        let redirection_data = services::RedirectForm::Form {
+            endpoint: item.response.pay_url.clone(),
+            method: services::Method::Post,
+            form_fields: Default::default(),
+        };
+        Ok(Self {
+            status: enums::AttemptStatus::AuthenticationPending,
+            response: Ok(types::PaymentsResponseData::TransactionResponse {
+                resource_id: types::ResponseId::ConnectorTransactionId(
+                    item.data.attempt_id.clone(),
+                ),
+                redirection_data: Some(redirection_data),
+                mandate_reference: None,
+                connector_metadata: None,
+                network_txn_id: None,
+            }),
+            ..item.data
+        })
+    }
+}
+
+impl<F, T>
+    TryFrom<
+        types::ResponseRouterData<
+            F,
+            CashtocodePaymentsSyncResponse,
+            T,
+            types::PaymentsResponseData,
+        >,
+    > for types::RouterData<F, T, types::PaymentsResponseData>
+{
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(
+        item: types::ResponseRouterData<
+            F,
+            CashtocodePaymentsSyncResponse,
+            T,
+            types::PaymentsResponseData,
+        >,
+    ) -> Result<Self, Self::Error> {
+        Ok(Self {
+            status: enums::AttemptStatus::Charged,
```

Comment:
why is the Charged value hardcoded? we can have processing and failed status also in PSync right?

Reply:
This is to the handle response of the webhook. Webhook is only received when it succeeds.